packages/devtools_server/lib/src/memory_profile.dart

```diff
@@ -194,8 +200,64 @@ class MemoryProfile {
     _pollingTimer = Timer(updateDelay, _pollMemory);
   }
 
-  Future<AdbMemoryInfo> _fetchAdbInfo() async =>
-      AdbMemoryInfo.fromJson((await getAdbMemoryInfo()).json);
+  /// Poll ADB meminfo
+  Future<AdbMemoryInfo> _fetchAdbInfo() async => AdbMemoryInfo.fromJson(
+        (await getAdbMemoryInfo()).json,
+        rawData: true,
+      );
+
+  /// Poll Fultter engine's Raster Cache metrics.
+  /// @returns engine's rasterCache estimates or null.
+  Future<RasterCache> _fetchRasterCacheInfo(IsolateRef selectedIsolate) async {
+    final response = await getRasterCacheMetrics(selectedIsolate);
+    if (response == null) return null;
+    final rasterCache = RasterCache.parse(response.json);
+    return rasterCache;
+  }
+
+  /// @returns view id of selected isolate's 'FlutterView'.
+  /// @throws Exception if no 'FlutterView'.
+  Future<String> getFlutterViewId(IsolateRef selectedIsolate) async {
+    final flutterViewListResponse = await callService(
+      registrations.flutterListViews,
+      isolateId: selectedIsolate.id,
+    );
+    final List<dynamic> views =
+        flutterViewListResponse.json['views'].cast<Map<String, dynamic>>();
+
+    // Each isolate should only have one FlutterView.
+    final flutterView = views.firstWhere(
+      (view) => view['type'] == 'FlutterView',
+      orElse: () => null,
+    );
+
+    if (flutterView == null) {
+      final message =
+          'No Flutter Views to query: ${flutterViewListResponse.json}';
+      print('ERROR: $message');
```

Comment:
remove print. Also what happens when this is run on a dartvm app? Does this code path get triggered?

Reply:
Done